src/main/kotlin/no/item/xp/plugin/util/NodeParserHelpers.kt

```diff
@@ -0,0 +1,43 @@
+package no.item.xp.plugin.util
+
+import arrow.core.Option
+import arrow.core.Some
+import no.item.xp.plugin.extensions.getAttributeAsOption
+import no.item.xp.plugin.extensions.getChildNodeAtXPath
+import no.item.xp.plugin.extensions.getChildNodesSequence
+import no.item.xp.plugin.extensions.getTextContentAsOption
+import no.item.xp.plugin.models.InputType
+import no.item.xp.plugin.models.YesNoMaybe
+import org.w3c.dom.Node
+
+fun getTypeForGeneratedField(node: Node): Option<InputType> =
+  node.getAttributeAsOption("type")
+    .map {
+      InputType.valueOf(it.toUpperCase())
+    }
```

Comment:
This is not important, but it might be even more  readable like this:
```suggestion
  node.getAttributeAsOption("type")
    .map{ it.toUpperCase() }
    .map(InputType::valueOf)
```